libc-bottom-half/sources/environ.c

```diff
@@ -10,9 +10,9 @@
 // that it's initialized whenever user code might want to access it.
 char **__wasilibc_environ;
 extern __typeof(__wasilibc_environ) _environ
-    __attribute__((weak, alias("__wasilibc_environ")));
+    __attribute__((__weak__, alias("__wasilibc_environ")));
 extern __typeof(__wasilibc_environ) environ
-    __attribute__((weak, alias("__wasilibc_environ")));
```

Comment:
I'm not opposed to changing these `weak`s to `__weak__`s, though (a) could you split this out into a separate PR, and (b) it's worth noting that most of these aren't header files, so so don't strictly need most of these.